src/board/BoardLogic.java

```diff
@@ -256,6 +257,13 @@ protected void killSnake(int snakeIndex) {
         this.snakesLocation.remove(snakeIndex);
         this.snakes.remove(snakeIndex);
 
+        // check if all existing apples are still valid
+        Set<Field> validAppleFields = getValidAppleFields();
+        for (Field appleField : getApples()) {
+            if (!validAppleFields.contains(appleField)) {
+                removeApple(appleField.getPosX(), appleField.getPosY());
+            }
```

Comment:
This iteration is not necessary if we pre compute this during the apple search function as discussed earlier

Reply:
Hash sets consume 2*N memory which can easily be avoided. Furthermore hash set operations are more expensive than array lockups

Reply:
Do you mean the boolean 2D array?
If so we would need two flood functions, the other still returning the array of valid fields as to not potentially check for a valid field on the 2D array forever.
All of that for a miniscule performance boost, I don't know

Reply:
I thought about it some more, there are certainly better ways to implement this, i.e. saving the accessible apples in a set instead of the  the valid fields. But I'm honestly too lazy to do that right now :joy: